Failing test due to new ipython version

The `tests/test_viewers.py::test_several_data_viewers` started failing recently with this stack trace: (e.g. [here](https://github.com/aiidalab/aiidalab-widgets-base/actions/runs/14314859668/job/40470183661))

```python
aiidalab_widgets_base/viewers.py:60: in viewer
    return _viewer(obj, **kwargs)
aiidalab_widgets_base/viewers.py:1639: in __init__
    output_notebook(hide_banner=True)
/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/bokeh/io/output.py:124: in output_notebook
    run_notebook_hook(notebook_type, "load", resources, verbose, hide_banner, load_timeout)
/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/bokeh/io/notebook.py:357: in run_notebook_hook
    return _HOOKS[notebook_type][action](*args, **kwargs)
/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/bokeh/io/notebook.py:475: in load_notebook
    publish_display_data({

    def publish_display_data(data: Dict[str, Any], metadata: Dict[Any, Any] | None = None,
            source: str | None = None, *, transient: Dict[str, Any] | None = None, **kwargs: Any) -> None:
        '''
    
        '''
        # This import MUST be deferred or it will introduce a hard dependency on IPython
        from IPython.display import publish_display_data
>       publish_display_data(data, metadata, source, transient=transient, **kwargs)
E       TypeError: publish_display_data() takes from 1 to 2 positional arguments but 3 positional arguments (and 1 keyword-only argument) were given

/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/bokeh/io/notebook.py:487: TypeError
```

Pinning ipython version <9.0 fixes this. I'll submit a PR.

Since the call to the ipython `publish_display_data` comes from bokeh, this could probably be fixed by updating bokeh as well, but that's a riskier proposition so we might do that later.

--- a/aiidalab_widgets_base/bug_report.py
+++ b/aiidalab_widgets_base/bug_report.py
@@ -11,11 +11,11 @@ import base64
 import json
 import platform
 import re
+import subprocess
 import sys
+import textwrap
 import zlib
-from subprocess import run
-from textwrap import wrap
-from urllib.parse import urlencode, urlsplit, urlunsplit
+from urllib import parse
 
 import ipywidgets as ipw
 from ansi2html import Ansi2HTMLConverter
@@ -25,7 +25,7 @@ def find_installed_packages(python_bin: str | None = None) -> dict[str, str]:
     """Return all currently installed packages."""
     if python_bin is None:
         python_bin = sys.executable
-    output = run(
+    output = subprocess.run(
         [python_bin, "-m", "pip", "list", "--format=json"],
         encoding="utf-8",
         capture_output=True,
@@ -185,13 +185,15 @@ def install_create_github_issue_exception_handler(output, url, labels=None):
                     # Determine and format the environment fingerprint to be
                     # included with the bug report:
                     environment_fingerprint="\n".join(
-                        wrap(get_environment_fingerprint().decode("utf-8"), 100)
+                        textwrap.wrap(
+                            get_environment_fingerprint().decode("utf-8"), 100
+                        )
                     ),
                 ),
                 "labels": ",".join(labels),
             }
-            issue_url = urlunsplit(
-                urlsplit(url)._replace(query=urlencode(bug_report_query))
+            issue_url = parse.urlunsplit(
+                parse.urlsplit(url)._replace(query=parse.eurlencode(bug_report_query))
             )
 
             with output:

--- a/setup.cfg
+++ b/setup.cfg
@@ -35,7 +35,7 @@ install_requires =
     spglib>=1.14,<3
     vapory~=0.1.2
     pandas~=2.1
-    ipython>=7.33
+    ipython>=7.33,<9.0
 python_requires = >=3.9
 include_package_data = True
 zip_safe = False

